Assets/Scripts/EnemyController.cs

```diff
@@ -109,9 +109,9 @@ void FindNavTarget()
         {
             Transform closestPlayer = players[0].transform;
             //find player closest to enemy
-            for(int i = 0; i < players.Length; i++)
+            for (int i = 0; i < players.Length; i++)
             {
-                if(Vector3.Distance(players[i].transform.position, transform.position) < Vector3.Distance(closestPlayer.position, transform.position)) //We can add in DDA here by multiplying the distances based on the player with a multiplier
+                if (players[i].GetComponent<PlayerManager>().health > 0 && Vector3.Distance(players[i].transform.position, transform.position) < Vector3.Distance(closestPlayer.position, transform.position)) //We can add in DDA here by multiplying the distances based on the player with a multiplier
                 {
                     closestPlayer = players[i].transform;
                 }
```

Comment:
This part might have a bug, though I'm not 100% sure. When you make closestPlayer as the first one in array, you don't check for his health. Then when you have to compare it, health comes into play, but maybe `players[0]` had `0` health and couldn't be picked?

Reply:
Also there's no need to start a for loop from 0 and compare it to itself at first. Though regarding my previous comment, this need might change

Reply:
great catch!

Reply:
Yeah it does have a bug, when I though of it a little more. So if stunned player is first in array and closer, monsters always consider it as the one to follow